src/libs/type-utils/lodash-fp-helpers.ts

```diff
@@ -0,0 +1,118 @@
+import _ from 'lodash/fp'
+import { AnyFn, AnyPromiseFn, GenericFn, WrapFn } from 'src/libs/type-utils/general-types'
+
+/*
+ * The helpers in this file are to provide even better type safety then the baseline lodash-fp utils
+ * can give when the scenario is known and specific, or a scenario presents additional type-safety
+ * challenges that the baseline type utils can't fully give type safety for.
+ */
+
+export interface WithHandlersFn {
+  <P, A extends any[], F extends (...args: A) => Promise<P>>(
+      handlers: WrapFn<(...args: A) => Promise<P | unknown>>[],
+      mainFn: F
+  ): F;
+  <F extends AnyFn, F2 extends F>(
+      handlers: WrapFn<GenericFn<F2>>[],
+      mainFn: F
+  ): GenericFn<F>
+}
+
+/**
+ * Provides better type flow and ergonomics for the
+ * common case of wrapping a main function with additional pass-thru wrapper
+ * functions (handlers).  These handlers are expected to be HoF's
+ * (higher order functions) that return a wrapped function with the desired
+ * handling logic which still honors the same function signature of the fn
+ * it is given as it's single argument
+ * @param handlers array of handler (wrapper) functions to apply to mainFn
+ * @param mainFn the main function being wrapped with handlers
+ */
+export const withHandlers : WithHandlersFn = <F extends AnyFn>(handlers: WrapFn<F>[], mainFn: F) => {
+  const resultFn = _.flow(...handlers)(mainFn)
+  return resultFn
+}
+
+export interface CurryLastArgFn {
+  <A, LAST, R>(
+      fn: (a: A, last: LAST) => R
+  ) : (a: A) => (last: LAST) => R;
+  <A, B, LAST, R>(
+      fn: (a: A, b: B, last: LAST) => R
+  ) : (a: A, b: B) => (last: LAST) => R;
+  <A, B, C, LAST, R>(
+      fn: (a: A, b: B, c: C, last: LAST) => R
+  ) : (a: A, b: B, c: C) => (last: LAST) => R;
+  <A, B, C, D, LAST, R>(
+      fn: (a: A, b: B, c: C, d: D, last: LAST) => R
+  ) : (a: A, b: B, c: C, d: D) => (last: LAST) => R;
+}
+
+/**
+ * produces a curried function that expect the last argument to be given
```

Comment:
Nit:
```suggestion
 * Produces a curried function that expect the last argument to be given
```

Similarly in a few other places

Reply:
fixed